src/authorizationProvider.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react";
+import authService from "./services/authService";
+
+const refreshToken = 'RefreshToken';
+const accessToken = 'AccessToken';
+const AuthorizationContext = React.createContext();
+class AuthorizationProvider extends Component {
+    state = {
+        loggedIn: null,
+        userFirstName: null,
+        userLastName: null,
+        userRole: null
+    }
+
+    async componentDidMount() {
+        if (localStorage.getItem(accessToken) !== 'null' &&
+            localStorage.getItem(refreshToken) !== 'null' &&
+            localStorage.getItem(accessToken) !== null &&
+            localStorage.getItem(refreshToken) !== null) {
+            const response = await authService.refresh(localStorage.getItem(accessToken), localStorage.getItem(refreshToken));
+
+            if (response.data.succeeded === true) {
+                let jwtData = response.data.accessToken.split('.')[1];
```

Comment:
je to asi rovnaky kod ako v `login()` metode nizsie, asi by sa to zislo dat do nejakej jednej novej funkcie.

Reply:
done